File: frontend/src/types/timesheet.ts
export type TimesheetStatus = 'draft' | 'submitted' | 'approved' | 'locked';

export interface Timesheet {
  id: string;
  user_id: string;
  user_name?: string;
  period_start: string;
  period_end: string;
  status: TimesheetStatus;
  submitted_at?: string;
  approved_by?: string;
  approved_at?: string;
  notes?: string;
  total_hours?: number;
  entry_count?: number;
  created_at: string;
  updated_at: string;
}

export interface CreateTimesheetRequest {
  period_start: string;
  period_end: string;
  notes?: string;
}

export interface UpdateTimesheetRequest {
  period_start?: string;
  period_end?: string;
  notes?: string;
}

export interface TimesheetListParams {
  page?: number;
  per_page?: number;
  search?: string;
  status?: string;
  user_id?: string;
}

File: backend/crates/konto-api/src/dto/timesheet.rs
use serde::{Deserialize, Serialize};
use utoipa::ToSchema;

#[derive(Debug, Serialize, ToSchema)]
pub struct TimesheetResponse {
    pub id: String,
    pub user_id: String,
    pub period_start: String,
    pub period_end: String,
    pub status: String,
    pub submitted_at: Option<String>,
    pub approved_by: Option<String>,
    pub approved_at: Option<String>,
    pub notes: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Deserialize, ToSchema)]
pub struct CreateTimesheetRequest {
    pub period_start: String,
    pub period_end: String,
    pub notes: Option<String>,
}

#[derive(Debug, Deserialize, ToSchema)]
pub struct UpdateTimesheetRequest {
    pub period_start: Option<String>,
    pub period_end: Option<String>,
    pub notes: Option<Option<String>>,
}

impl From<konto_db::entities::timesheet::Model> for TimesheetResponse {
    fn from(m: konto_db::entities::timesheet::Model) -> Self {
        Self {
            id: m.id,
            user_id: m.user_id,
            period_start: m.period_start.to_string(),
            period_end: m.period_end.to_string(),
            status: m.status,
            submitted_at: m.submitted_at.map(|t| t.to_string()),
            approved_by: m.approved_by,
            approved_at: m.approved_at.map(|t| t.to_string()),
            notes: m.notes,
            created_at: m.created_at.to_string(),
            updated_at: m.updated_at.to_string(),
        }
    }
}
